File: src/engine/transpile/src/test/resources/golden/SYK005/WS_編集メッセージ.py
"""SYK005 のデータ部レコード WS-編集メッセージ を逐語対訳した自動生成コード。"""
from cobol_runtime import (
    decode_alphanumeric,
    decode_binary,
    decode_packed,
    decode_zoned,
    encode_alphanumeric,
    encode_binary,
    encode_packed,
    encode_zoned,
)


class WS_編集メッセージ:
    """01 WS-編集メッセージ 総バイト長 100。"""

    LENGTH = 100

    def __init__(self, data=None):
        if data is None:
            self._data = bytearray(100)
        else:
            self._data = bytearray(data)
            if len(self._data) < 100:
                self._data.extend(b"\x00" * (100 - len(self._data)))

    @property
    def data(self):
        return self._data

    # WS-編集メッセージ
    def get_WS_編集メッセージ(self):
        return decode_alphanumeric(self._data, 0, 100)
    def set_WS_編集メッセージ(self, value):
        encode_alphanumeric(self._data, 0, 100, value)

File: src/engine/transpile/src/test/resources/golden/SYK005/WS_編集メッセージ.java
package cobolinsight.generated;

import cobolinsight.runtime.CobolRuntime;
import java.util.Arrays;

/** SYK005 のデータ部レコード WS-編集メッセージ を逐語対訳した自動生成コード。 */
public final class WS_編集メッセージ {

    /** 01 WS-編集メッセージ 総バイト長 100。 */
    public static final int LENGTH = 100;

    private final byte[] data;

    public WS_編集メッセージ() {
        this.data = new byte[100];
    }

    public WS_編集メッセージ(byte[] source) {
        this.data = Arrays.copyOf(source, 100);
    }

    public byte[] data() {
        return data;
    }

    // WS-編集メッセージ
    public String get_WS_編集メッセージ() {
        return CobolRuntime.decodeAlphanumeric(data, 0, 100);
    }
    public void set_WS_編集メッセージ(String value) {
        CobolRuntime.encodeAlphanumeric(data, 0, 100, value);
    }
}
